server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -708,6 +710,39 @@ public void testNumericLongOrDateSortOptimization() throws Exception {
         dir.close();
     }
 
+    public void testIndexFieldHasDuplicateData() throws IOException {
+        final int numDocs = 10000;
+        final int threshold1 = numDocs * 60 / 100;
+        final int threshold2 = numDocs * 40 / 100;
+        final int threshold3 = numDocs * 5 / 100;
+
+        final String fieldName = "duplicateField";
+        final String fieldName2 = "notMuchDuplicateField";
+        final String fieldName3 = "notDuplicateField";
+
+        long duplicateValue = randomLongBetween(-10000000L, 10000000L);
+        long value, value2, value3;
+        Directory dir = newDirectory();
+        IndexWriter writer = new IndexWriter(dir, new IndexWriterConfig(null));
+        for (int i = 0; i < numDocs; ++i) {
+            value = i < threshold1 ? duplicateValue : i;
+            value2 = i < threshold2 ? duplicateValue : i;
+            value3 = i < threshold3 ? duplicateValue : i;
+            Document doc = new Document();
+            doc.add(new LongPoint(fieldName, value));
+            doc.add(new LongPoint(fieldName2, value2));
+            doc.add(new LongPoint(fieldName3, value3));
+            writer.addDocument(doc);
+        }
+        writer.close();
+        final IndexReader reader = DirectoryReader.open(dir);
+        assertTrue(indexFieldHasDuplicateData(reader, fieldName));
```

Comment:
We could also check the value returned to compute the heuristic (the median and the count) and compare it with the expectation with some error bounds. It could be another test but if we generate values randomly we should be able to compute the median and the number of duplicated values accurately and then compare the result with the approximation that is computed in the BKD, wdyt ?